src/main/java/com/epam/esm/repository/Repository.java

```diff
@@ -0,0 +1,15 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.entities.Identifiable;
+
+import java.util.List;
+
+public interface Repository<T extends Identifiable> {
+
+    void save(T entity);
+    T getById(int id);
+    List<T> getAll();
+    void update(T entity);
+    void delete(int id);
+
```

Comment:
Are u sure, that you need all this methods to work with Tag and GiftCertificate?
According to your task - no. 
![image](https://user-images.githubusercontent.com/49719217/132389813-24af5129-38d4-43ab-b311-3cd4ba16535f.png)

You may to create parameterized BaseRepository<T> and declare common methods in it.
Also, you will need to create a separate interfaces, such as GiftCertificateRepository, TagRepository, which will inherit BaseRepository and expand the specific functionality